### ISSUE TITLE
noise sur generateur LFO

### ISSUE THREAD

**Issue by pach** [ossia/score#707]
Would be great to be able to adjust noise speed. For exemple in noise if we adjust frequency, it will recalculate noise on different tick. Or would change a coefficient on tick (virtual) time...

**Comment by bltzr** [comment]
Couldn't the effect you want (and more) be achieved with a low pass filter?

**Comment by jln-** [comment]
mm... What Thomas describe sounds more like a noise generator ran through a sample and hold function, doesnt' it ?

**Comment by pach** [comment]
mm...
This issue is really about the noise generator inside the LFO in the last version of @jcelerier, as discussed quickly with @jcelerier on the gitter.
The noise is not affected by the frequency (which affect sinus, and all other signal generation of this interval rack) and too fast for the effects I'm searching to build with. ;)

**Comment by bltzr** [comment]
Yes, we got this - we just proposed two different ways to solve the problem: 
- what @jln- suggests is to add a “Hold” function (like in an analog synth’s Sample&Hold function, or like Hold in Chataigne’s automations)
- what I suggest is a lo-pass filter (such as the max patcher below) that would give the sensation of a smoother noise, without changing the rate at which data is delivered
Of course, the best would be to combine both

Try the patcher below, and tell us what you think 

<pre><code>
----------begin_max5_patcher----------
2194.3oc6bsrbqaiDcs7WAGtH0jY70E.36ralUYextjTpfDgjQBEgFRHecRp
7uOfnIk.kDo.kHkcb4ptkuV.DO5C5Gfcej+yGl4tP7Jqz046b9ImYy9yGlMS
2TUCyp+7L2MzWWlQK0Ol6RwlMrbo6iPeR1qRc6+ykB1puyA4HEN3uso67caD
6jYLodvHiAsTjIJfEF8TbHIxO3wpeKBEDGC+VruG9QG7SHmeodfaKXkpEmJ4
h74ErkRXBvQwOgdzwCG7jdRP5ereTkxeOio2jFaKddytB2L4T4xm44qMlXBI
BlXR0+AqBwbp4o54Ur3W+henaUa+0COT8iGsDNyYeUM3SPyLAMcCqrzI.gNO
V1rqglj+9VFrkcc6QtelmlxxMGtsHANHnR3wgnJHFizfLgzATDbMPQWZVqJX
+u6r9Th9jFm3WqJMV5S6mXREJ5g6UexeLAwBpjcmAwXsFSX3XigfYXX.XY1K
D5Mhljac9gu46GnsHWcRXqH60sHG5Yn0DG0qoGw0bdXE0BZsjNycEOi8BqnT
cXYr0m4R2t0n4YFCoBd9UPsH5w8Mwygl712TA6Edy3822JsPIMRknrq.D9Wi
CcOLMhTVQ9Ntdq.MpNndnYBa.fnDMBDocCSBzpTdAdGf.0Y55VJuF5bszVm4
xR4xJ38nA3QB7SpddO+.ThVY2KBZC2dcxDK+MVpgoi5feKKmmaZHzp6T1J5t
L47UhbYI+Oz3fNjxY5eEcIqyAmS2.f3+ofSyZfQ20E7TQd0ln0QZUyMKWkGb
MTDXJL5mHmt8LCVoLof+N5rTIj6JWPKpNwW.Z0jlNkBQV6t1KKYrUx5t2xyy
OBEkhsc2YAe8y8L1EBUma5at08TNeWNz6bk8tbdI8k16PIMKq1AP6o+UZNei
xIpjCGADz9NY4Tkf9b4xBQVVKbB54kyzSpxXYI6q7T4y5ExTYP837sMJQt6O
kS4qYkx1sIoqKa2xAuLuvxdgIOzwtE0tDlKYa1lUEPP6tS7i+24ozhey0PXO
bmPSGAldNa0duAgNJPDaKiJcT+S9LyQYzrS5XDbp6.T2TPpdCTg.2JfW1iiT
cdW2cFwpynVHOXQ.GRmF25XG4QtMMqCeU6YbBNPbJoJsAl933EZ1t6wIwHfm
XHLHVGK.iBtDdFNl3YqqHzFN4cCelhTuWUnOs0.jQLvqUYkzO3B20uVWsVUk
zMz5emfVoyBmdfWRev6BZ9Z2GOCLe6phAvsyBhz2oGcIzx6NgV+aG7XoJd6Z
TX8c1iAGsIU.UePD4NAQqUwcbH8gS8pSUoOMM.EFtrVf+kTlv+cyq1siPglN
98Bun4VzcBh1vjEBk8O5JgJsGpwVaBY5aBdk4dM7RFSvRuMuiJNHahpA2a.b
S60y8E7hMWmlKPc75yySYuZ7xqeDwsFkHaAtjAAbj2I.2niZ9ICA07QCB07F
QTCfB6t0+HbwcHU8Hrk3B9lMCOjtiLddWuXqdmV0+4ArRwthkMpJMtGbZuqU
u3tjmuO6O+TSDZkRfyuX040P2D9VtGp.wpGbR1Dp.rVtKlPjva.6AzDtGvVr
Gz.1TADXa2DSHPfeOnVV4P0pcg5NOS2wgdxQ1pTLYZlICAKZ+fhhTnHBnoAf
Ffaz2d3oazo+.OzkKUgsFV5xdvrXHtUIpNcNjT94TorfuXmDBlYVcmqYgtpD
OuNSrflUmP48kk3xYg9PBqe3.ZMR03LS7kspmwYEOSpNUtqU6zmTWap3Qtbm
j52ZCJ6Wcw+5ndmdgiYIiqyK7237rHK8dilPZcC8GczDtUdPhQoT6BMG0Bv+
ytEzu9ytN4BdIyYMKmUPkh8ZoUtqVJ1A2skLMf8M.aQ9Gx1RD7JMdcfZjqA0
VkIT6kCkttP4QQYCOGJfloze17yc9by0JiJcppE.ZDPQfrPUaknXCUO3vqjC
MPc7C5kBMjnqAFkh0qyXCGFsi.Ccffd.gLrGAsEuRPF1pDHWn9c.WwiHSOJn
4ohMNXR7DR2CxEo6ARS0pnvdUSlDRmgdJ48AoyHlw9t.oyHAiJYeJ2HDxmGH
LnbiQGQUfZdGF1ub6+QjwOvIODsNzWC.9wnOI7ymD94SB+79ivOst9V8sjNn
4m7DNDgUlXlIFtyqlbMUg8zZm0swS2ow9ja1M7ja6qcXi8gjbWecu93BfYkD
K3kKo0XwSI2mpV9O9hxa2.N3FPEUZGL7pOUFT8Nqw+5KKFeI7eTYtxmFAMTy
J135ZW1HH3M2H3eoVJRvGFqff52aBtCYxkvebh68Cli93.yI3AAyQ2Mdeg93
3PGx6DNF3fXzkv3X2OnrI.Lo8Q1UzbB9ukbWYzofADHxZTi7lQbk6KEL.iJR
X3ABq1Kt389jBF1x9gorT2ACi2A2u5pZ6FK5D9BzaIMu8MlsrEIdBqJe7..m
1O2Fd5VAOWVqBG6oSMMBhRAEsp8mZdiDHYt5OcOUCrkWL3ojDD3jAblOUVpD
7PfhIaWXKeGBlvMgsp+DuO46v6A9N71UbzfZWZXuou3nddG3nd+0FE68VThY
0jw2nlfJwnI4sUEEoow8Yz0F7dTQNRfQwhfuozcBcWU04Ky3oGHTyHTV4KWU
d38o8hGaBfTWXsH.kP6euxyBVn2Yf0sJ1DOaD6j2eLO3lOuIMjGnaJDLlURu
46Bz.Ke7k7QLrpGCtRO8K8yoeS6FnPuklyx5mPXW+AF70lfT+GzGHEaXbGbM
BMQa+N7JQ7AlY.4zdD8JU+cEo9OrEWRp8tyRcs0C7UYYzkZRThgTG0EY7Zwt
C8SzNWB.Fb7UgqQhSycPq2Otq6+ZvI6SPaaWIhMKT3HrPXOaVoCul2UuPgVt
N2LxYyB4OJqTrsqzshcjHaWIxMtR91n3kLFfmMlR3w3XJwFEO+wPj7sQjPig
QKwVYpslG3.7HFbUsHGwbqiXs0oL1pa1ZcLSsTq7e8v+Gbh147C
-----------end_max5_patcher-----------
</code></pre>

**Comment by jcelerier** [comment]
I think that both way make sense in the LFO object: in particular, the low-pass could be a "smooth" attribute, and sample-and-hold would be a specific "noise" kind, like sin / square / noise1..3 / etc .

**Comment by jln-** [comment]
> I think that both way make sense in the LFO object: in particular, the low-pass could be a "smooth" attribute, and sample-and-hold would be a specific "noise" kind, like sin / square / noise1..3 / etc .

🤘 🎛

**Comment by bltzr** [comment]
wouldn’t that be even better to have a general purpose S&H (that would also apply on other functions), and, YES, adding a smooth attribute would be great!

I gave it a try but still cannot figure out how all this cords-business works...

**Comment by jcelerier** [comment]
sure thing (about the S&H).

in the meantime here's an approximative script that does it (not sample accurate but oh well): 

```QML
import QtQuick 2.0
import Score 1.0

Item {
  ValueInlet { id: in1 }
  ValueOutlet { id: out1 }
  IntSlider { id: sl; min: 10; max: 10000; }

  property var val: 0;
  property int count: 0;

  function onTick(oldtime, time, position, offset) 
  {
    var prev = count % sl.value;
    count += (time - oldtime);
    if((count % sl.value) < prev)  
    { 
      count = 0;
      val = in1.value;
    }
           
    out1.value = val;
  }
}
```

**Comment by jcelerier** [comment]
Implemented a S&H in 3a3eb72820e9a548cc9806c513fd0bce96252d67
